cms/djangoapps/contentstore/views/helpers.py

```diff
@@ -271,6 +281,75 @@ def create_xblock(parent_locator, user, category, display_name, boilerplate=None
         return created_block
 
 
+class ImportIdGenerator(IdGenerator):
+    """
+    Modulestore's IdGenerator doesn't work for importing single blocks as OLX,
+    so we implement our own
+    """
+    def __init__(self, context_key):
+        super().__init__()
+        self.context_key = context_key
+
+    def create_aside(self, definition_id, usage_id, aside_type):
+        """ Generate a new aside key """
+        raise NotImplementedError()
+
+    def create_usage(self, def_id) -> UsageKey:
+        """ Generate a new UsageKey for an XBlock """
+        # Note: Split modulestore will detect this temporary ID and create a new block ID when the XBlock is saved.
+        return self.context_key.make_usage_key(def_id.block_type, LocalId())
+
+    def create_definition(self, block_type, slug=None) -> DefinitionLocator:
+        """ Generate a new definition_id for an XBlock """
+        # Note: Split modulestore will detect this temporary ID and create a new definition ID when the XBlock is saved.
+        return DefinitionLocator(block_type, LocalId(block_type))
+
+
+def import_staged_content_from_user_clipboard(parent_key: UsageKey, user):
+    """
+    Import a block (and any children it has) from "staged" OLX.
+    Does not deal with permissions or REST stuff - do that before calling this.
+
+    Returns the newly created block on success or None if the clipboard is
+    empty.
+    """
+    if not content_staging_api:
+        raise RuntimeError("The required content_staging app is not installed")
+    user_clipboard = content_staging_api.get_user_clipboard_status(user.id)
+    if (
+        not user_clipboard or
+        user_clipboard.content.status != content_staging_api.StagedContentStatus.READY
+    ):
+        # Clipboard is empty or expired/error/loading
+        return None
+    source_usage_key = user_clipboard.source_usage_key  # TODO: track this somewhere
```

Comment:
Where should I store this? Simplest option would be to use a global XBlock mixing to add a new field to all XBlocks that can hold this value. More complex would be to store it in [split modulestore's `edit_info.original_usage`](https://github.com/openedx/edx-platform/blob/62e59d8aceb7e82fa49eb490b6b3509b03c66324/xmodule/modulestore/split_mongo/split.py#L1476-L1489) but that would require a new API, and not be forwards-compatible with Learning Core.

Reply:
Yeah, I was definitely thinking an XBlock mixin field, like we already do for other common ones.

Reply:
Ok, that's what I've done.